update/update-from-2.16-to-2.17.php

```diff
@@ -0,0 +1,170 @@
+ *
+ * IMPORTANT: they must be ordered for their proper execution.
+ */
+$sqlFiles = array();
+$sqlFiles[] = SQLPATH . "update-project-add-customer-relation.sql";
+
+// function inspired by code from install/setup-config.php
+function parse_psql_dump($url,$nowhost,$nowport,$nowdatabase,$nowuser,$nowpass){
+	$link = pg_connect("host=$nowhost port=$nowport user=$nowuser dbname=$nowdatabase password=$nowpass");
+	if (!$link) {
+		return false;
+	}
+
+	$file_content = file($url);
+	$string = "";
+	$success = true;
+	foreach($file_content as $sql_line){
+		$string = $string . $sql_line;
+		if(trim($string) != "" && strstr($string, "--") === false){
+			if (strstr($string, "\\.") != false)
+			{
+				pg_put_line($link, $string);
+				pg_end_copy($link);
+				$string = "";
+			} elseif (strstr($string, ";") != false)
+			{
+				if (!pg_query($link, $string)) {
+					$success = false;
+				}
+				$string = "";
+			}
+		} else $string = "";
+	}
+
+	return $success;
+}
+
+function check_db_version($nowhost,$nowport,$nowdatabase,$nowuser,$nowpass){
+	$link = pg_connect("host=$nowhost port=$nowport user=$nowuser dbname=$nowdatabase password=$nowpass");
+	if (!$link) {
+		return false;
+	}
+
+	$result = pg_query($link, "select version from config");
+
+	if ($result != NULL) {
+		$version = pg_fetch_array($result);
+		if (strcmp($version["version"], "2.16") == 0) {
+			return true;
+		}
+	}
+	return false;
+}
+// run upgrade scripts
+
+require_once(PHPREPORT_ROOT . 'config/config.php');
+
+
+if (!check_db_version(DB_HOST,DB_PORT,DB_NAME,DB_USER,DB_PASSWORD)) {
+	print ("Wrong database version. " .
+		"Make sure DB is on 2.16 version before running this upgrade.\n");
+	exit();
+}
+
+
+
+require_once(PHPREPORT_ROOT . 'config/config.php');
+include_once(PHPREPORT_ROOT . '/util/DBPostgres.php');
+
+if (!check_db_version(DB_HOST,DB_PORT,DB_NAME,DB_USER,DB_PASSWORD)) {
+	print ("Wrong database version. " .
+		"Make sure DB is on 2.1 version before running this upgrade.\n");
+	exit();
+}
+
+$success = true;
+foreach ($sqlFiles as $file) {
+	if (!parse_psql_dump($file,DB_HOST,DB_PORT,DB_NAME,DB_USER,DB_PASSWORD)) {
+		$success = false;
+		break;
+	}
+}
+// finish, print message
+
+if ($success) {
+
+	$link = pg_connect( "host=" . DB_HOST . " port=" . DB_PORT . " user=" . DB_USER . " dbname=" . DB_NAME . " password=" . DB_PASSWORD );
+
+	if ( !$link ) {
+		error_log( "ERROR: Could not connect to DB" );
+	}
+
+	$result = pg_query( $link, $query = "SELECT customerid, projectid FROM requests" );
+	if ( $result == null ) {
+		error_log( "ERROR: Could not run query: $query" );
+	}
+
+	$customerAssignedToProject = array();
+
+	while ( $row = pg_fetch_array( $result, null, PGSQL_ASSOC ) ) {
+		$customerAssignedToProject[$row['projectid']][] = $row['customerid'];
+	}
+
+	foreach ( $customerAssignedToProject as $projectId => $value ) {
+		$customerAssociated = $value[0];
+		if ( count( $customerAssignedToProject[$projectId] ) > 1 ) {
```

Comment:
You can probably use `count($value)` here. Even, instead of `$value`, you could call it `$customerList` for the code to be more readable.